File: 프로그래머스/unrated/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.java
class Solution {
    public String solution(String my_string, int[][] queries) {
        StringBuilder sb = new StringBuilder(my_string);

		for (int[] q : queries) {
			StringBuilder nsb = new StringBuilder(sb.substring(q[0], q[1] + 1)).reverse();
			sb.replace(q[0], q[1] + 1, nsb.toString());
		}

		return sb.toString();
    }
}

File: 프로그래머스/unrated/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.kt
class Solution {
    fun solution(my_string: String, queries: Array<IntArray>): String {
        var ans = my_string

        queries.forEach { (s, e) ->
            ans = ans.slice(0 until s) + ans.slice(s .. e)
                .reversed() + ans.slice(e + 1 .. ans.lastIndex)
        }

        return ans
    }
}